api/v1alpha1/docker_clusterconfig_types.go

```diff
@@ -4,56 +4,15 @@
 package v1alpha1
 
 import (
-	"maps"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 
 	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/openapi/patterns"
 )
 
-//+kubebuilder:object:root=true
-
-// DockerClusterConfig is the Schema for the dockerclusterconfigs API.
-type DockerClusterConfig struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec DockerClusterConfigSpec `json:"spec,omitempty"`
-}
-
 type DockerSpec struct {
 	GenericClusterConfig `json:",inline"`
 }
 
-// DockerClusterConfigSpec defines the desired state of DockerClusterConfig.
-type DockerClusterConfigSpec struct {
-	// +optional
-	Docker *DockerSpec `json:"docker,omitempty"`
-
-	//+optional
-	CustomImage *OCIImage `json:"customImage,omitempty"`
```

Comment:
Wait, I messed up here with the `CustomImage`.

Reply:
I tested this before I pushed so it's really interesting that this didn't break anything re usage via patches, etc :sweat_smile: